### PR TITLE
Feature/notifications toast and snackbar

### PR DESCRIPTION
**Pull request by kostasdano**
## Description

This is the third part of the implementation of the issue #58

- Introducing **Toast**: **Generic Toast** and **Toast Notification**. This feature is implemented with the **Toast** and **NotificationVisual** components

- Introducing **Snackbar Notification**, implemented with the **Snackbar** component

## Screenshot
[Abstract link](https://app.abstract.com/projects/96260c1d-f169-4306-ab88-2472d8e457fd/branches/master/commits/latest/files/d7cb764d-aa66-4fd6-90dd-4804917a3ed3/layers/336D3D69-30DA-4DD5-B855-5F18BBCE962A?collectionId=ae1fabbe-9c71-457e-8959-f7fee7411487&collectionLayerId=45bd2b6b-0673-4f71-a107-89161bf8b103)

- Toast

toast-generic-final

- Toast Notification (Toast and NotificationVisual)

toast-notification

- Snackbar Notification

snackbar

## Test Plan

**Toast Tests**

- Generic Toast with simple visual renders correctly

- Generic Toast with simple visual expands/shrinks correctly

- Generic Toast with simple visual closeCTA works properly

**Toast-NotificationVisual Tests**

- Toast with NotificationVisual renders correctly

- Toast with NotificationVisual expands/shrinks correctly

- Toast with NotificationVisual closeCTA works properly

- Toast with NotificationVisual primaryCTA and secondaryCTA work properly

**Snackbar Tests**

- Snackbar Notification renders correctly

- Snackbar Notification's closeCTA works properly

- Snackbar Notification's primaryCTA and secondaryCTA work properly

### CODE REVIEW

File: src/components/Notification/Notification.test.tsx [hunk 1]
```diff
@@ -119,3 +123,174 @@ describe('Notifications Container', () => {
...
+
+    const { findByTestId, findByText } = render(
+      <Toast {...toastData} closeCTA={closeCTA}>
+        <NotificationVisual {...visualData} primaryCTA={primaryCTA} secondaryCTA={secondaryCTA} />
+      </Toast>
+    );
+
+    // Toast is initialized with expanded=false
+    // Note: The NotificationVisual is rendered regardless of the isExpanded value, so it can be always found in document
+    // If isExpanded=true, the min-height of the NotificationVisual is 0, else > 0.
```
Comment (SVonKarajann) [inline src/components/Notification/Notification.test.tsx:171]: probably meant to write isExpanded=false here

**Comment by kostasdano** [comment]
you-get-a-5bace9

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -1,17 +1,16 @@
 ---
```
Comment (panvourtsis) [inline .github/ISSUE_TEMPLATE/bug_report.md:1]: why these files are changed?

File: src/components/Toast/Toast.style.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { Theme } from '../../theme';
+import { rem, transparentize } from 'polished';
+import { css, SerializedStyles } from '@emotion/core';
+import { transition } from '../../theme/functions';
+import { ToastType } from './Toast';
+import { AcceptedColorComponentTypes } from '../../utils/themeFunctions';
+import { isNotificationTypes } from './Toast';
+
+const isAcceptedComponentTypes = (type: string): type is AcceptedColorComponentTypes => {
+  return [
```
Comment (panvourtsis) [inline src/components/Toast/Toast.style.ts:10]: use the `mainTypes` from palette.js here

File: src/components/Toast/Toast.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import { TestId } from '../../utils/types';
+import { generateTestDataId } from '../../utils/helpers';
+
+export type ToastType = AcceptedColorComponentTypes | string;
+
+export type Props = {
+  /** The informative message of the Toast */
+  message: string;
+  /** The type of the Toast, will determine the color and the icon */
+  type: ToastType;
```
Comment (panvourtsis) [inline src/components/Toast/Toast.tsx:28]: why this is `AcceptedColorComponentTypes | string` ?

Comment (kostasdano) [reply src/components/Toast/Toast.tsx:28]: my initial thought was that the user may want to use a Toast type different than the ones in AcceptedComponentTypes (for now only AcceptedComponentTypes are accepted) and therefore a default color (darkGray) is set for this (designs also include a default color/no icon version of the toast)

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (kostasdano) [reply .github/ISSUE_TEMPLATE/bug_report.md:1]: tbh I have no clue (not manually changed, maybe due to update or yarn?)

File: src/components/Toast/Toast.tsx (hunk 4 above)
Comment (panvourtsis) [reply src/components/Toast/Toast.tsx:28]: there is no other type though so i would stick with what we know so no string for now

File: src/components/Toast/Toast.style.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  color: ${theme.palette.white};
+  display: flex;
+  justify-content: space-between;
+  overflow: hidden;
+  height: ${rem(58)};
+  background: ${theme.utils.getColor(type, 400, 'normal')};
+`;
+
+export const infoContainer = () => (theme: Theme): SerializedStyles => css`
+  display: flex;
```
Comment (panvourtsis) [inline src/components/Toast/Toast.style.ts:42]: optional: we have css function for centered flex -> `flexCenter`

File: src/components/Toast/Toast.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  /** The informative message of the Toast */
+  message: string;
+  /** The type of the Toast, will determine the color and the icon */
+  type?: AcceptedColorComponentTypes;
+  /** The closing call-to-action of the Toast */
+  closeCTA: (() => void) | undefined;
+  /** Initialize toast as expanded */
+  expanded?: boolean;
+  /** Children of the Toast */
+  children: React.ReactNode | React.ReactNode[] | undefined;
```
Comment (panvourtsis) [inline src/components/Toast/Toast.tsx:32]: as this is under `React.FC<>` children prop is automaticaly provided

**Review by panvourtsis — APPROVED** [review]

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**